### PR TITLE
Stop polluting global namespace with CT

### PR DESCRIPTION
**Pull request by keradus**
Practically done.
2 non-functional changes still to be done (marked as TODO in code)

### CODE REVIEW

File: src/Console/Command/FixCommand.php [hunk 1]
```diff
@@ -287,8 +286,6 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        Transformers::create(); // make sure all transformers have defined custom tokens since configuration might depend on it
```
Comment (keradus) [inline src/Console/Command/FixCommand.php]: that was the couse of whole refactoring

**Comment by SpacePossum** [comment]
:+1: awesome :)

File: src/Tokenizer/CT.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * @internal
```
Comment (keradus) [inline src/Tokenizer/CT.php]: actually not, it may be used in 3rd party fixers

**Comment by GrahamCampbell** [comment]
I'm actually not so sure this is needed? Why does this matter if this is just a tool?

**Comment by GrahamCampbell** [comment]
Surely it's just making branch merges really hard now?

**Comment by keradus** [comment]
To not pollute global namespace with some constants at one, quite random runtime execution point.
So basically fix https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2227#discussion_r84184603

**Pull request closed without merging**